gazelle/README.md

```diff
@@ -199,6 +199,43 @@ Python-specific directives are as follows:
 | `# gazelle:resolve py ...` | n/a |
 | Instructs the plugin what target to add as a dependency to satisfy a given import statement. The syntax is `# gazelle:resolve py import-string label` where `import-string` is the symbol in the python `import` statement, and `label` is the Bazel label that Gazelle should write in `deps`. | |
 
+
+#### Directive: `python_root`:
+
+Set this directive within the Bazel package that you want to use as the Python root.
+For example, if using a `src` dir (as recommended by the [Python Packaging User
+Guide][python-packaging-user-guide]), then set this directive in `src/BUILD`:
+
+```starlark
+# ./src/BUILD
+# Tell gazelle that are python root is the same dir as this Bazel package.
+# gazelle:python_root
+```
+
+Note that the directive does not have any arguments.
+
+Gazelle will then add `imports = [".."]` (or similar) to all targets that it
+generates:
+
+```starlark
+# in ./src/foo/BUILD
+py_libary(
+    ...
+    imports = [".."],  # Gazelle adds this
+    ...
+)
+
+# in ./src/foo/bar/BUILD
+py_libary(
+    ...
+    imports = ["../.."],  # Gazelle adds this
+    ...
+)
+```
+
+[python-packaging-user-guide]: https://packaging.python.org/en/latest/tutorials/packaging-projects/
```

Comment:
Agreed, but sadly it doesn't look like the project maintains any other versions: https://readthedocs.org/projects/python-packaging-user-guide/

I could switch to the latest doc build (they build docs on all PRs), but:

1. I don't know how long ReadTheDocs keeps PR builds. The earliest PR build was PR 1000 https://readthedocs.org/projects/python-packaging-user-guide/builds/?page=107.
2. It's possible that the PR linked is reverted in the future.

The docs link for the most recent docs build is https://python-packaging-user-guide--1511.org.readthedocs.build/en/1511/ and would have this in the bottom corner:

![image](https://github.com/bazelbuild/rules_python/assets/5386897/c107bc98-ad3e-405c-99ce-350f57436b9d)

Other options:
1. Remove the link and the "as recommended by" comment altogether.
2. Link to the `pytest` docs instead which would allow me to pin the version (https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html#tests-outside-application-code). However, those docs focus more on where to put tests rather than telling people to use a `src` dir.

Reply:
You can reference the git commit that produced the doc: https://github.com/pypa/packaging.python.org/blob/4c86169a/source/tutorials/packaging-projects.rst

Reply:
Thanks to GitHub, it renders the doc nicely, and it becomes versioned!

Reply:
Love it. Updated!